packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -305,56 +305,50 @@ module Floats = {
     /*
       This function goes through a sorted array and divides it into two different clusters:
       continuous samples and discrete samples. The discrete samples are stored in a mutable map.
-      Samples are thought to be discrete if they have any duplicates.
+      Samples are thought to be discrete if they have at least `minDiscreteWight` duplicates.
+
+      If the min discreet weight is 4, that would mean that at least four elements needed from a specific
+      value for that to be kept as discrete. This is important because in some cases, we can expect that
+      some common elements will be generated by regular operations. The final continous array will be sorted.
  */
-    let _splitContinuousAndDiscreteForDuplicates = (sortedArray: array<float>) => {
+    let splitContinuousAndDiscreteForMinWeight = (
+      sortedArray: array<float>,
+      ~minDiscreteWeight: int,
+    ) => {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
-      Belt.Array.forEachWithIndex(sortedArray, (index, element) => {
-        let maxIndex = (sortedArray |> Array.length) - 1
-        let possiblySimilarElements = switch index {
-        | 0 => [index + 1]
-        | n if n == maxIndex => [index - 1]
-        | _ => [index - 1, index + 1]
-        } |> Belt.Array.map(_, r => sortedArray[r])
-        let hasSimilarElement = Belt.Array.some(possiblySimilarElements, r => r == element)
-        hasSimilarElement
-          ? FloatFloatMap.increment(element, discrete)
-          : {
-              let _ = Js.Array.push(element, continuous)
+
+      let flush = (cnt: int, value: float): unit => {
+        if cnt >= minDiscreteWeight {
+          FloatFloatMap.add(value, cnt->Belt.Int.toFloat, discrete)
+        } else {
+          for _ in 1 to cnt {
```

Comment:
Is there no way you can't just construct an array first with ``value``, then push it? Like, ``E.make(ctn, value)->Js.Array2.push``?

If there's nothing official, we could make our own, like, ``mutateAppend``.

Reply:
There is `Js.Array2.pushMany`. But it would be a bit slower, since we'd have to build the array first and then append it.

In the most common case samples are continuous, and `cnt` is 1. (I know this is probably not the bottleneck, O(N) vs O(N*log(N)) win is much more important here, but still.)

Reply:
Good point. If that's the case, maybe just add a comment on that last line above this. I'd agree this might give us a minor improvement if cnt is often 1. (I'd expect that otherwise though, it would be a bit slower, if Array.make() and pushMany are decently performant)

Reply:
Huh. In my experience, in general, for/while loops and plain values are significantly faster than any function calls and objects (including arrays).

I'm not sure if this is always true in JS, though, with its crazy JIT optimizations, it could heavily rewrite and inline stuff.